src/shared/cloudformation/activation.ts

```diff
@@ -48,15 +46,100 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
                 getIdeProperties().codelenses
             )
         )
-        getLogger().error('Failed to activate template registry', e)
+        getLogger().error('Failed to activate template registry: %s', e)
         // This prevents us from breaking for any reason later if it fails to load. Since
         // Noop watcher is always empty, we will get back empty arrays with no issues.
         globals.templateRegistry = new NoopWatcher() as unknown as CloudFormationTemplateRegistry
     }
     // If setting it up worked, add it to subscriptions so it is cleaned up at exit
     extensionContext.subscriptions.push(
-        globals.templateRegistry,
         Commands.register('aws.cloudFormation.newTemplate', () => createStarterTemplateFile(false)),
         Commands.register('aws.sam.newTemplate', () => createStarterTemplateFile(true))
     )
 }
+
+/**
+ * Sets the `templateRegistry` property in the `globals` variable.
+ *
+ * This does the following:
+ * - Shows a 'loading' message to the user if the template registry variable is
+ *   requested but not yet available. For example, this would be triggered if
+ *   template file is opened.
+ * - Returns the correct registry object once all background loading is completed.
+ */
+function setTemplateRegistryInGlobals(registry: CloudFormationTemplateRegistry, disposables: vscode.Disposable[]) {
+    Object.defineProperty(globals, 'templateRegistry', {
```

Comment:
why is this dynamic patching needed?  can we make the control flow more obvious (maybe something static that lives on `CloudFormationTemplateRegistry` as @JadenSimon suggested)